include/nbl/builtin/hlsl/numeric_limits.hlsl

```diff
@@ -0,0 +1,202 @@
+// Copyright (C) 2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_NUMERIC_LIMITS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_NUMERIC_LIMITS_INCLUDED_
+
+#include <type_traits.hlsl>
+
+using half = double;
```

Comment:
this needs to go into the `ifndef` portion of `cpp_compat` or similar, and use `float16_t` instead of `half`

Then we either use `std::float16_t` from `<stdfloat>` if @AnastaZIuk can bump Nabla to C++23 or you need to switch between compilers:
- Clang & GCC `_Float16`
- MSVC add a git mirror of https://half.sourceforge.net/ as a submodule

Or alterantively take a dive inside our OpenEXR submodule, update it to latest, see if the EXR/image tests still run
https://github.com/AcademySoftwareFoundation/openexr/blob/159e57120dae5e587454e0d4bd916c792831a8ce/website/PortingGuide.rst#L7

You'll need help from @AnastaZIuk to upgrade the CMake build system to add the spliced out `IlmBase` submodule to NAbla's 3rdparty

Then there's `IlmBase/Half/half.h` inside which looks more or less C++23 compliant and even has a std::numeric_limits specialization for C++

Reply:
ok even with C++23, the support for individual `std::floatN_t` is optional, looks like Clang doesn't implement (but at least has a `_Float16` you can use), GCC does and waiting for confimarion about MSVC 19.38 and 19.39